stdlib/public/libexec/swift-backtrace/main.swift

```diff
@@ -540,6 +539,69 @@ Generate a backtrace for the parent process.
       currentThread = target!.crashingThreadNdx
     }
 
+    // Set up the output stream
+    var didOpenOutput = false
+    switch args.outputTo {
+      case .stdout:
+        outputStream = standardOutput
+      case .stderr:
+        outputStream = standardError
+      case .file:
+        if isDir(args.outputPath) {
+          // If the output path is a directory, generate a filename
+          let name = target!.name
+          let pid = target!.pid
+          var now = timespec()
+
+          if clock_gettime(CLOCK_REALTIME, &now) != 0 {
+            now.tv_sec = time(nil)
+            now.tv_nsec = 0
+          }
+
+          var filename =
+            "\(args.outputPath)/\(name)-\(pid)-\(now.tv_sec).\(now.tv_nsec).log"
+
+          var fd = open(filename, O_RDWR|O_CREAT|O_EXCL, 0o644)
+          var ndx = 1
+          while fd < 0 && errno == EEXIST {
```

Comment:
You might want to also check for `EINTR` here in case someone resizes their window at exactly the wrong moment. A repeat/while loop would let you avoid repeating the filename/open logic but that's just a style thing.

Reply:
Good point.